### ISSUE TITLE
[Litestar] nested not working with create_pydantic_model response

### ISSUE THREAD

**Issue by ricksore** [piccolo-orm/piccolo#860]
I have this table:
```
class Profile(Table):
    class Role(str, Enum):
        STAFF = 'staff'
        SUPERVISOR = 'supervisor'
        ADMIN = 'admin'

    user = ForeignKey(references=BaseUser)
    user = ForeignKey(references=Company)
```

and created the Pydantic model
```
ProfileModel = create_pydantic_model(
    Profile,
    nested=(Profile.user)
)
```

and this endpoint
```

**Comment by ricksore** [comment]
Okay I think the problem is with the `to_dict()` method; apparently it also converts FKs into dictionaries.

**Comment by sinisaos** [comment]
@ricksore I'm sorry, but I can't reproduce your problem. If I do this, everything is fine,  
```python
class Company(Table):
    name = Varchar()

class Profile(Table):
    class Role(str, Enum):
        STAFF = "staff"
        SUPERVISOR = "supervisor"
        ADMIN = "admin"

    user = ForeignKey(references=BaseUser)
    company = ForeignKey(references=Company)
    role = Varchar(choices=Role)
    activated = Timestamp()


RegisterData = create_pydantic_model(
    Profile,
    model_name="RegisterData",
)
ProfileModel = create_pydantic_model(
    Profile,
    include_default_columns=True,
    model_name="ProfileModel",
    nested=(Profile.user),
)

@post("/profiles", status_code=201, tags=["Profile"])
async def create_profile(data: RegisterData) -> ProfileModel:
    profile = Profile(**data.dict())
    await profile.save()
    return profile.to_dict()
```
and response is
```javascript
{
  "id": 1,
  "user": 1,
  "company": 1,
  "role": "staff",
  "activated": "2023-07-10T07:53:00"
}
```
Can you please post what the json response should look like in your case?

**Comment by dantownsend** [comment]
One thing to be careful with is this:

```python
ProfileModel = create_pydantic_model(
    Profile,
    nested=(Profile.user)
)
```

Make sure there's a trailing comma, otherwise it's not treated as a tuple, so should be:

```python
ProfileModel = create_pydantic_model(
    Profile,
    nested=(Profile.user,)  # <- Note here
)
```

And with Litestar, I'm not sure if it automatically converts the response to the Pydantic model. You can try doing it explicitly instead:

```python
    @post('register', status_code=201)
    async def register(self, data: RegisterData) -> ProfileModel:
        ...
        profile = Profile(user=user, company=company, activated=datetime.now())
        await profile.save()
        
        return ProfileModel(**profile.to_dict())  # <- Note here
```

**Comment by sinisaos** [comment]
> Make sure there's a trailing comma, otherwise it's not treated as a tuple, so should be:
> 
> ```python
> ProfileModel = create_pydantic_model(
>     Profile,
>     nested=(Profile.user,)  # <- Note here
> )
> ```

Yes. You're right, I forgot to put a comma when I pasted the code.

> And with Litestar, I'm not sure if it automatically converts the response to the Pydantic model. You can try doing it explicitly instead:
> 
> ```python
>     @post('register', status_code=201)
>     async def register(self, data: RegisterData) -> ProfileModel:
>         ...
>         profile = Profile(user=user, company=company, activated=datetime.now())
>         await profile.save()
>         
>         return ProfileModel(**profile.to_dict())  # <- Note here
> ```
Neither FastAPI nor Litestar work that way. Sorry if I don't understand the problem, but if we want nested response output (after POST data and save to database) the only way I managed without getting a `ValidationError`
```shell
pydantic.error_wrappers.ValidationError: 1 validation error for ProfileModel
user
  value is not a valid dict (type=type_error.dict)
```
is to make another query with nested output like this:
```python
@post("/profiles", status_code=201, tags=["Profile"])
async def create_profile(data: RegisterData) -> ProfileModel:
    profile = Profile(**data.dict())
    await profile.save()
    saved_profile = profile.to_dict()
    profile = (
        await Profile.select(
            Profile.all_columns(),
            Profile.user.all_columns(),
            Profile.company,
        )
        .where(Profile.id == saved_profile["id"])
        .first()
        .output(nested=True)
    )
    return ProfileModel(**profile)
```
![nested_output](https://github.com/piccolo-orm/piccolo/assets/30960668/37d391a0-5e9e-416b-8eaf-bbdb2f63b896)


I apologize again if I missed the point.

**Comment by dantownsend** [comment]
> Neither FastAPI nor Litestar work that way

I think FastAPI supports this in recent versions. You had to do this in the past:

```python
@app.get('/', response_model=MyModel)
def my_endpoint():
    ...
```

But now I think this works:

```python
@app.get('/')
def my_endpoint() -> MyModel:
    ...
```

> is to make another query with nested output like this:

You're right - that works. You can do something similar with objects:

```python
# Gets the nested user object:
profile = await Profile.objects(Profile.user).where(Profile.id == some_id).first()
return ProfileModel(**profile.to_dict())
```

Or lets say you already have the profile, without the nested user object you can get it like this:

```python
profile.user = await profile.get_related(Profile.user)
return ProfileModel(**profile.to_dict())
```

**Comment by sinisaos** [comment]
> I think FastAPI supports this in recent versions. You had to do this in the past:
> 
> ```python
> @app.get('/', response_model=MyModel)
> def my_endpoint():
>     ...
> ```
> 
> But now I think this works:
> 
> ```python
> @app.get('/')
> def my_endpoint() -> MyModel:
>     ...
> ```
> 

Unfortunately neither method works without additional nested queries with `fastapi==0.99.1` which is the last version before Pydantic V2.

> 
> You're right - that works. You can do something similar with objects:
> 
> ```python
> # Gets the nested user object:
> profile = await Profile.objects(Profile.user).where(Profile.id == some_id).first()
> return ProfileModel(**profile.to_dict())
> ```
> 
> Or lets say you already have the profile, without the nested user object you can get it like this:
> 
> ```python
> profile.user = await profile.get_related(Profile.user)
> return ProfileModel(**profile.to_dict())
> ```

Both of these methods **work perfectly** in FastAPI or Litestar and they are much cleaner than the query builder `select` method (I almost always use `select` method (query builder), but here it is much better to use `objects` method (ORM)). 
@ricksore Is this what you need for an response?

**Comment by sinisaos** [comment]
@ricksore [to_dict()](https://github.com/piccolo-orm/piccolo/blob/97292ccc27c379e17e9e964100a9da804055430b/piccolo/table.py#L690) is just a helper method for converting an object to a dictionary. The best way to achieve what you want is, as @dantownsend  advised, to let Piccolo find the associated object and pass it to a result like this.
```python
profile.user = await profile.get_related(Profile.user) 
return ProfileModel(**profile.to_dict())
```
With that one line of code, you get a nicely nested response.
